### ISSUE TITLE
ModuleNotFoundError: No module named 'mediapipe.python._framework_bindings' on Raspberry Pi 3

### ISSUE THREAD

**Issue by goulaoalex** [google/mediapipe#3130]
Hello, I'm having problems using mediapipe on my raspberry pi 3. 
Using "import mediapipe" gives no error, however using "mp_drawing = mp.solutions.drawing_utils" (for example) gives out the following error message:
ModuleNotFoundError: No module named 'mediapipe.python._framework_bindings'

**My installation method was:**
1. sudo apt install ffmpeg python3-opencv python3-pip ;
2. sudo apt install libxcb-shm0 libcdio-paranoia-dev libsdl2-2.0-0 libxv1  libtheora0 libva-drm2 libva-x11-2 libvdpau1 libharfbuzz0b libbluray2 libatlas-base-dev libhdf5-103 libgtk-3-0 libdc1394-22 libopenexr25 ;
3. sudo pip3 uninstall mediapipe-rpi3 .

I'm using a raspberry pi 3b with debian bullseye (32 bit), my python version is 3.9.2 and the opencv version is 4.2.1. 
P.S. the file name is "maos.py".

Does anyone know what might be causing this error?
(I annexed an image of the error aswell for clarity
![Error](https://user-images.githubusercontent.com/71357969/156027739-24788cb9-43a0-416e-868f-25e73edbe877.JPG)
)

**Comment by goulaoalex** [comment]
I've also experienced the error "AttributeError: module 'mediapipe' has no attribute 'solutions'" after uninstalling and instaling mediapipe.
![Error 2](https://user-images.githubusercontent.com/71357969/156029167-f642050d-c0cd-4fe3-9ff5-07d24d394986.JPG)

**Comment by sureshdagooglecom** [comment]
Hi @goulaoalex ,
Thanks for raising issue ,did you get a chance to check this issue https://github.com/google/mediapipe/issues/1346

**Comment by goulaoalex** [comment]
I checked the issue and followed the steps from "https://github.com/superuser789/MediaPipe-on-RaspberryPi#readme"

But while building mediapipe with "python3 setup.py bdist_wheel" I got the following error message:

/home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/com_google_absl/absl/base/BUILD.bazel:201:11: Compiling absl/base/internal/cycleclock.cc failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 44 arguments skipped)

Do you know what might be causing this?

**Comment by goulaoalex** [comment]
Further note:
My gcc version is 10.2.1

**Comment by goulaoalex** [comment]
Further note:
"gcc is already the newest version (4:10.2.1-1)."

**Comment by yoyojacky** [comment]
I have the same issue too. 
OS : Raspberry Pi OS 64bit bullseye.
Kernel Version : 5.10.92-v8+
GCC version : 10.2.1 20210110(Debian 10.2.1-6)

ENV:
virtualenv environment : 
mediapipe-rpi4 ==0.8.8 
opencv-python==4.5.5.62
opencv-contrib-python==4.5.5.62
numpy==1.22.2

and encount the same issue when i try to print multi_hand_landmarks..
![mediapipeerr](https://user-images.githubusercontent.com/6893075/157037051-847e8543-d99f-4a5d-8c6b-34640748939a.png)

**Comment by yoyojacky** [comment]
> 

Hi goulaoalex, 
I am going to try it out, thanks for your tips...

**Comment by tedbryantesoro** [comment]
> Hey yoyojacky I followed the installation method from https://google.github.io/mediapipe/getting_started/install.html and I got it to work!
> 
> However I wasn't able to install bazelisk so I followed the following method to install bazel from source:
> 
> sudo apt-get install openjdk-11-jdk
> 
> sudo nano /etc/dphys-swapfile 2048 sudo /etc/init.d/dphys-swapfile stop sudo /etc/init.d/dphys-swapfile start
> 
> wget https://github.com/bazelbuild/bazel/releases/download/5.0.0/bazel-5.0.0-dist.zip unzip -d bazel bazel-5.0.0-dist.zip cd bazel
> 
> sudo nano scripts/bootstrap/compile.sh -c Line 144 -J-Xmx1024M
> 
> env EXTRA_BAZEL_ARGS="--host_javabase=@local_jdk//:jdk" bash ./compile.sh sudo cp output/bazel /usr/local/bin/bazel
> 
> I hope this can help you.

hello i followed your method with a bit of researching on some details, i get to install bazel however the error still occured mediapipe.framework binding error, is there any solution u have in mind? thank u

**Comment by goulaoalex** [comment]
Hey, when you type "import mediapipe" in python does it give out the framework error or does it load instantly? 
Also, while building mediapipe with bazel did you get any errors?

**Comment by tedbryantesoro** [comment]
> Hey, when you type "import mediapipe" in python does it give out the framework error or does it load instantly? Also, while building mediapipe with bazel did you get any errors?

i did, as shown on the image that i uploaded mediapipe.python._framework_bindings error

**Comment by goulaoalex** [comment]
What about the mediapipe bazel build, did you get any errors?

**Comment by tedbryantesoro** [comment]
> What about the mediapipe bazel build, did you get any errors?

what do you mean? should i suppose to build mediapipe with bazel? because i haven't done that yet

**Comment by goulaoalex** [comment]
Yep, from my experience installing mediapipe with pip3 doesn't work on raspberry pi 3. Follow the "Installing on Debian and Ubuntu" tutorial from this page https://google.github.io/mediapipe/getting_started/install.html#installing-on-debian-and-ubuntu

If you need help with any steps let me know.

**Comment by tedbryantesoro** [comment]
> Yep, from my experience installing mediapipe with pip3 doesn't work on raspberry pi 3. Follow the "Installing on Debian and Ubuntu" tutorial from this page https://google.github.io/mediapipe/getting_started/install.html#installing-on-debian-and-ubuntu
> 
> If you need help with any steps let me know.

thank u, i actually had problems following some steps from the site that u gave, it would be helpful

**Comment by goulaoalex** [comment]
In which steps did you have problems?

**Comment by tedbryantesoro** [comment]
> In which steps did you have problems?

hey i just did the steps from that site, i followed the steps accordingly and even done the config for opencv. That's the option i did too( _Option 2. Run [setup_opencv.sh](https://github.com/google/mediapipe/blob/master/setup_opencv.sh) to automatically build OpenCV from source and modify MediaPipe’s OpenCV config. This option will do all steps defined in Option 3 automatically._ ) however, it still gave me the same error

**Comment by goulaoalex** [comment]
After that follow this steps:
https://google.github.io/mediapipe/getting_started/python.html#building-mediapipe-python-package

**Comment by tedbryantesoro** [comment]
> After that follow this steps: https://google.github.io/mediapipe/getting_started/python.html#building-mediapipe-python-package

thank u, but what does it mean here (_4. In the virtual environment, go to the MediaPipe repo directory._)

**Comment by goulaoalex** [comment]
In the installation, you probably cloned the mediapipe repository with "git clone https://github.com/google/mediapipe.git"

So going to the mediapipe repo directory is just to "cd mediapipe"

**Comment by tedbryantesoro** [comment]
> In the installation, you probably cloned the mediapipe repository with "git clone https://github.com/google/mediapipe.git"
> 
> So going to the mediapipe repo directory is just to "cd mediapipe"

hey, thanks for helping me so far im at step 5 rn (_Install the required Python packages.

(mp_env)mediapipe$ pip3 install -r requirements.txt_) my only concern is im at the building wheel process for opencv-contrib-python (PEP517) i heard there's an issue about this being stuck for long, i experienced that from numpy it took awhile to install so i wonder if you just waited for it to finish?

**Comment by goulaoalex** [comment]
Yes the building wheel process took a long time for me too, I usually wait for it to build (if it doesn't give out an error). 
If the build fails trying upgrading the setup tools with this:

pip install --upgrade pip setuptools wheel

> > In the installation, you probably cloned the mediapipe repository with "git clone https://github.com/google/mediapipe.git"
> > So going to the mediapipe repo directory is just to "cd mediapipe"
> 
> hey, thanks for helping me so far im at step 5 rn (_Install the required Python packages.
> 
> (mp_env)mediapipe$ pip3 install -r requirements.txt_) my only concern is im at the building wheel process for opencv-contrib-python (PEP517) i heard there's an issue about this being stuck for long _(EDIT: It's been running for a bit long now),_ i experienced that from numpy it took awhile to install so i wonder if you just waited for it to finish?

**Comment by goulaoalex** [comment]
Yes the building wheel process took a long time for me too, I usually wait for it to build (if it doesn't give out an error).
If the build fails trying upgrading the setup tools with this:

pip install --upgrade pip setuptools wheel

**Comment by tedbryantesoro** [comment]
> 

So uhm the update is, ive been running my rpi for like 4+ hours installing opencv and i've noticed it still haven't completed installing yet, i tried upgrading using that  command line too,

**Comment by goulaoalex** [comment]
If that's the case then it's probably not gonna work. I don't remember having problems with opencv-contrib.
Maybe try this with this command: 

sudo apt install python3-opencv

It's all I used to install opencv

**Comment by tedbryantesoro** [comment]
> 

i tried running that command and this was the out put
python3-opencv is already the newest version (4.5.1+dfsg-5).

**Comment by goulaoalex** [comment]
I don't know the solution for the building wheel problem, I've also experienced it a couple of times and sometimes it worked and others it didn't.
My advice would be to either let the building wheel process run overnight or try building opencv from source since you also download the source code for opencv-contrib.

Here's a tutorial that I know works for the opencv installation from source (it works for raspberry pi 3):
https://singleboardbytes.com/647/install-opencv-raspberry-pi-4.htm

This process takes a while (the make command usually took me around 4 hours), but if you don't have any other solution give it a shot.

**Comment by tedbryantesoro** [comment]
> I don't know the solution for the building wheel problem, I've also experienced it a couple of times and sometimes it worked and others it didn't. My advice would be to either let the building wheel process run overnight or try building opencv from source since you also download the source code for opencv-contrib.
> 
> Here's a tutorial that I know works for the opencv installation from source (it works for raspberry pi 3): https://singleboardbytes.com/647/install-opencv-raspberry-pi-4.htm
> 
> This process takes a while (the make command usually took me around 4 hours), but if you don't have any other solution give it a shot.

Does it also works with Raspberry pi 3 b+ model?

**Comment by goulaoalex** [comment]
Yes I have the same model and it worked for me.

**Comment by tedbryantesoro** [comment]
> Yes I have the same model and it worked for me.

hello thank you, i successfully installed opencv since i thoroughly followed the instructions, i just noticed that i was using two diff. env the one from mediapipe (mp_env) and opencv (sb_cvv) from the instruction you sent, should i install the mediapipe to the (sb_cvv) environment? or should i redo the installation but this time to the mediapipe environment i was thinking of using thonny python too for coding

**Comment by goulaoalex** [comment]
Just install mediapipe to the sb_cvv environment so you don't have to redo the opencv installation

**Comment by tedbryantesoro** [comment]
> Just install mediapipe to the sb_cvv environment so you don't have to redo the opencv installation

im on this step rn
_5. Install the required Python packages.

(mp_env)mediapipe$ pip3 install -r requirements.txt_

hello, im just installing mediapipe requirements rn however it started building wheels for opencv-contrib-python again
![image](https://user-images.githubusercontent.com/85889960/158648267-8965dfdc-7746-4fb6-a559-fe9fab8d38ef.png)

**Comment by tedbryantesoro** [comment]
> 

Im doing step 6 (last step)
I encountered this error 

![image](https://user-images.githubusercontent.com/85889960/158671200-496eb7d5-be53-484a-9319-74a7033b8e9a.png)

**Comment by goulaoalex** [comment]
I've also experienced that error, the solution I found was to use an older version of gcc and gcc++
I used gcc 9.

Here's the tutorial I used to switch versions:
https://linuxconfig.org/how-to-switch-between-multiple-gcc-and-g-compiler-versions-on-ubuntu-20-04-lts-focal-fossa

I can't say 100% for sure that it will work, I was also stuck on that error for a while but I got to it work somehow with this.

**Comment by tedbryantesoro** [comment]
> I've also experienced that error, the solution I found was to use an older version of gcc and gcc++ I used gcc 9.
> 
> Here's the tutorial I used to switch versions: https://linuxconfig.org/how-to-switch-between-multiple-gcc-and-g-compiler-versions-on-ubuntu-20-04-lts-focal-fossa
> 
> I can't say 100% for sure that it will work, I was also stuck on that error for a while but I got to it work somehow with this.

Sadly, it didn't work for me i tried using auto and manual mode 
(sbb_cv) pi@raspberrypi:~/mediapipe $ python3 setup.py install --link-opencv
/home/pi/.virtualenvs/sbb_cv/lib/python3.9/site-packages/setuptools/dist.py:519: UserWarning: The version specified ('dev') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  warnings.warn(
running install
/home/pi/.virtualenvs/sbb_cv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running build_binary_graphs
generating binarypb: mediapipe/modules/face_detection/face_detection_short_range_cpu
DEBUG: Rule 'rules_foreign_cc' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "c2cdcf55ffaf49366725639e45dedd449b8c3fe22b54e31625eb80ce3a240f1e"
DEBUG: Repository rules_foreign_cc instantiated at:
  /home/pi/mediapipe/WORKSPACE:42:13: in <toplevel>
Repository rule http_archive defined at:
  /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/a87782c34d667d1c1a18fe82a9a7abfc72fd345b.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
WARNING: Download from http://mirror.tensorflow.org/github.com/bazelbuild/rules_closure/archive/cf1e44edb908e9616030cc83d085989b8e6cd6df.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/tensorflow/runtime/archive/fc70a32ecd248dd7612d75d2177125ed14541367.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'com_google_benchmark' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'pybind11_bazel' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'xctestrunner' because it already exists.
DEBUG: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/org_tensorflow/third_party/repo.bzl:122:14:
Warning: skipping import of repository 'pybind11' because it already exists.
INFO: Build option --define has changed, discarding analysis cache.
DEBUG: Rule 'rules_cc' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "3839996049629e6377abdfd04681ddeeb0cc3db13b9d2ff81bf46700cb4529f7"
DEBUG: Repository rules_cc instantiated at:
  /home/pi/mediapipe/WORKSPACE:36:13: in <toplevel>
Repository rule http_archive defined at:
  /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
WARNING: /home/pi/mediapipe/mediapipe/framework/tool/BUILD:182:24: in cc_library rule //mediapipe/framework/tool:field_data_cc_proto: target '//mediapipe/framework/tool:field_data_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
WARNING: /home/pi/mediapipe/mediapipe/framework/BUILD:54:24: in cc_library rule //mediapipe/framework:calculator_cc_proto: target '//mediapipe/framework:calculator_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
INFO: Analyzed target //mediapipe/modules/face_detection:face_detection_short_range_cpu (1 packages loaded, 5241 targets configured).
INFO: Found 1 target...
ERROR: /home/pi/.cache/bazel/_bazel_pi/b64ded70520c15cba5314bb754a27c61/external/XNNPACK/BUILD.bazel:6642:19: Compiling src/qu8-gavgpool/7p7x-minmax-scalar-c1.c failed: undeclared inclusion(s) in rule '@XNNPACK//:scalar_prod_microkernels':
this rule is missing dependency declarations for the following files included by 'src/qu8-gavgpool/7p7x-minmax-scalar-c1.c':
  '/usr/lib/gcc/arm-linux-gnueabihf/9/include/stddef.h'
  '/usr/lib/gcc/arm-linux-gnueabihf/9/include/stdint.h'
  '/usr/lib/gcc/arm-linux-gnueabihf/9/include/stdbool.h'
Target //mediapipe/modules/face_detection:face_detection_short_range_cpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 34.222s, Critical Path: 4.59s
INFO: 131 processes: 130 internal, 1 linux-sandbox.
FAILED: Build did NOT complete successfully

**Comment by goulaoalex** [comment]
Sadly I don't know the solution for that error and I don't know how to help you there, for me after that somehow it worked.

I was sent this video today on a post I made a few weeks ago on stackoverflow if you wanna check it out. I don't know if this method works or not but at least it's recent.

https://www.youtube.com/watch?v=vF6fwKN06lM

They are saying to use raspbian buster

**Comment by tedbryantesoro** [comment]
> Sadly I don't know the solution for that error and I don't know how to help you there, for me after that somehow it worked.
> 
> I was sent this video today on a post I made a few weeks ago on stackoverflow if you wanna check it out. I don't know if this method works or not but at least it's recent.
> 
> https://www.youtube.com/watch?v=vF6fwKN06lM
> 
> They are saying to use raspbian buster

Hello may i ask what python version were you using?, because i was using 3.9 while building mediapipe, i think that might've caused it because i think mediapipe does not support 3.9 or it doesn't matter

**"pip install mediapipe" does not work #1466**

**Comment by goulaoalex** [comment]
I was using version 3.9.2. 
Mediapipe does support newer versions than 3.6 and 3.8, I think that issue is outdated.

pip install mediapipe, however, doesn't work for raspberry for some reason, that's why someone created the "pip install mediapipe-rpi3" package. But I also think that said package doesn't work anymore for the newer versions of raspbian.

**Comment by goulaoalex** [comment]
I spent a lot of time switching between python version, I can assure it always gave me the same framework error.

**Comment by tedbryantesoro** [comment]
> /sysdeps/unix/sysv/linux/arm/sigrestorer.S:64

idk if it's like the required processor but i think that might be the cause

(sbb_cv) pi@raspberrypi:~/mediapipe $ uname -m
armv7l

i guess im using a 32 bit processor

**Comment by goulaoalex** [comment]
Honestly I don't know for sure, but armv7l is a 32bit processor. You are using the 64-bit debian version right?

**Comment by tedbryantesoro** [comment]
> Honestly I don't know for sure, but armv7l is a 32bit processor. You are using the 64-bit debian version right?
 Im using 32 bit

**Comment by goulaoalex** [comment]
I installed mine with the 64-bit version, I don't know if that's the cause of the problem or not but either way I got the same error as you in my version.

**Comment by yoyojacky** [comment]
64bit OS may not working properly,  i am using raspberry pi OS 64bit (codename:  bullseye) and it can not work properly. 
I have installed via:
```
pip install mediapipe-rpi4
```
but when i write demo code to test it. .... program crashed ....

**Comment by stefan-sherwood** [comment]
I had the same problem using Python 3.8.0 on Rpi4 (bullseye)

```
$ python
Python 3.8.0 (default, Mar 22 2022, 08:04:42)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mediapipe
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/stefan/hand_tracking/venv/lib/python3.8/site-packages/mediapipe/__init__.py", line 16, in <module>
    from mediapipe.python import *
  File "/home/stefan/hand_tracking/venv/lib/python3.8/site-packages/mediapipe/python/__init__.py", line 17, in <module>
    from mediapipe.python._framework_bindings import resource_util
ModuleNotFoundError: No module named 'mediapipe.python._framework_bindings'
```

I had installed mediapipe using  `pip install mediapipe-rpi4`

I solved it with `pip install mediapipe`

Now:

```
$ python
Python 3.8.0 (default, Mar 22 2022, 08:04:42)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mediapipe
>>>

```

**Comment by hutaohutc** [comment]
> 
when I use `pip install mediapipe`  in raspi4, 
get error
```
ERROR: Could not find a version that satisfies the requirement mediapipe
ERROR: No matching distribution found for mediapipe

```

**Comment by stefan-sherwood** [comment]
@hutaohutc What version of Python are you using?

**Comment by hutaohutc** [comment]
> @hutaohutc What version of Python are you using?

python 3.9

**Comment by stefan-sherwood** [comment]
@hutaohutc https://github.com/google/mediapipe/issues/1466#issuecomment-754964239

**Comment by Matesanz** [comment]
For me the problem **was solved** by **NOT runing the python console on the mediapipe project folder**.

```python
import mediapipe as mp
```

:point_right: **What you expect:** mediapipe is being imported from the environment installed dependencies
:point_right: **What actually happens:** it is trying to be imported from the `mediapipe/` subfolder located in the mediapipe project

**Comment by gabmiro** [comment]
Hi guys, I have the same exact problem. I am trying to solve it. What is the best method?

**Comment by stefan-sherwood** [comment]
@gabmiro People have reported several different problems in this report and all have been solved. You have to give specific details about your situation if you want a good answer.

**Comment by gabmiro** [comment]
@stefan-sherwood I am having the same issue has goulaoalex. The ModuleNotFoundError. I am trying his method to download Bazel but I can't get the compilation the work.

**Comment by stefan-sherwood** [comment]
@gabmiro What version of Python are you using? Note that the same error message does not mean the same issue. Please include what command you are issuing and everything that is output, including the error message. There are multiple different errors in this with different causes.

**Comment by gabmiro** [comment]
![crise](https://user-images.githubusercontent.com/123667100/218273619-1c9fd48e-1f28-48cd-bcbb-b73e94b03360.png)
 I am getting this error

**Comment by stefan-sherwood** [comment]
@gabmiro I'm a little confused. The error you showed has nothing about mediapipe in it. And that's what this whole report is about. That said, Python 3.9 is not yet supported for MediaPipe, as shown in [this bug report](https://github.com/google/mediapipe/issues/1466#issuecomment-754964239).

As a side note, building should never require sudo (installing, yes. building, no)

**Comment by gabmiro** [comment]
@stefan-sherwood I know that's why I tried to install Bazel like @goulaoalex did. So you think It is better I downgrade my python version instead of installing Bazel?

**Comment by stefan-sherwood** [comment]
@gabmiro I can't say what the best solution is for you because you haven't said what you're trying to accomplish, what you're doing to accomplish it (and with what software), or what error messages you're getting. The Bazel route just is a complication/distraction and totally unnecessary in most cases.

**Comment by gabmiro** [comment]
@stefan-sherwood I am trying to import mediapipe to my raspberry pi for a school project, I have never work with that before. I tried to downgrade my python version before but pip wasn't working with the downgraded version. Maybe you could help me.

**Comment by stefan-sherwood** [comment]
@gabmiro Different answer depending on which Raspberry Pi you are talking about. Also, what version is pip not working and what is the error output? Details!

**Comment by gabmiro** [comment]
I am using the Raspberry Pi 3B+. Do you have a link of instruction to downgrade my python version. I will try it and if I have error I will show you.

**Comment by stefan-sherwood** [comment]
I think you should search Stack Overflow for instructions on how to get multiple (or just a different) version of Python running on your machine. That topic has been covered quite extensively.

**Comment by gabmiro** [comment]
@stefan-sherwood I will try that again, if it doesn't work could you help me?

**Comment by gabmiro** [comment]
and which version of python should I get?

**Comment by stefan-sherwood** [comment]
The answer to your questions depend on what you're trying to do with MediaPipe.

For example (from https://google.github.io/mediapipe/getting_started/install.html):

"Note: To make Mediapipe work with TensorFlow, please set Python 3.7 as the default Python version and install the Python “six” library by running pip3 install --user six."

**Comment by gabmiro** [comment]
I am planning to use MediaPipe face mesh to detect sleepness

**Comment by gabmiro** [comment]
![crise2](https://user-images.githubusercontent.com/123667100/218285233-2bb042f3-297a-4d85-a0e9-3d64a4e16274.png)

I am getting this error whenever I try to use pip.

**Comment by gabmiro** [comment]
![crise3](https://user-images.githubusercontent.com/123667100/218286525-0c7bb80a-eab4-4815-a7e1-838c6616e935.png)
Alright so after a while, I was able to download Python 3.7 and make it the default version. I installed mediapipe and opencv like it is showed on this https://pypi.org/project/mediapipe-rpi4/ but unfortunalety I have the same error.

**Comment by stefan-sherwood** [comment]
How did you install mediapipe and what version is it?

**Comment by gabmiro** [comment]
@stefan-sherwood I have finally able to solve my problem, thank you for your help.

**Comment by stefan-sherwood** [comment]
@gabmiro That's great to hear. Mind sharing what the fix was?

**Comment by gabmiro** [comment]
@stefan-sherwood I downloaded python 3.7 but I don't know if this is mandatory and I use "python -m pip install mediapipe --user" this command instead of "pip install mediapipe-rpi3" and it works.
